exporters/otlp/trace/src/jmh/java/io/opentelemetry/exporter/otlp/trace/RequestMarshalBenchmarks.java

```diff
@@ -70,9 +71,18 @@
   @Threads(1)
   public byte[] marshalProtoCustom(RequestMarshalState state) throws IOException {
     ExportTraceServiceRequest protoRequest =
-        TraceMarshaler.RequestMarshaler.create(state.spanDataList).toRequest();
+        toRequest(TraceMarshaler.RequestMarshaler.create(state.spanDataList));
     byte[] protoOutput = new byte[protoRequest.getSerializedSize()];
     protoRequest.writeTo(CodedOutputStream.newInstance(protoOutput));
     return protoOutput;
   }
+
+  private static ExportTraceServiceRequest toRequest(TraceMarshaler.RequestMarshaler request)
+      throws IOException {
+    byte[] buf = new byte[request.getSerializedSize()];
+    request.writeTo(CodedOutputStream.newInstance(buf));
+    return ExportTraceServiceRequest.newBuilder()
+        .setUnknownFields(UnknownFieldSet.newBuilder().mergeFrom(buf).build())
```

Comment:
Wow, what's the performance boost for this?

Reply:
Bogdan originally wrote these benchmarks, but I think it's to compare serializing without having to vendor in the entire `MarshalerInputStream` / stub code you saw. But in the end going with the vendored in code had a significant improvement so we went with it (don't have the numbers on hand)